lib/native/aws_iot.ts

```diff
@@ -362,6 +472,18 @@ export class AwsIotMqttConnectionConfigBuilder {
             this.params.tls_ctx = new io.ClientTlsContext(this.tls_ctx_options);
         }
 
+        // Add the metrics string
+        if (this.params.username == undefined || this.params.username == null || this.params.username == "") {
```

Comment:
We use this or its negation quite a few times in here, maybe make a helper function.